test/WebJobs.Script.Tests.Integration/Management/InstanceControllerTests.cs

```diff
@@ -80,5 +82,67 @@ public async Task Assign_MSISpecializationFailure_ReturnsError()
             Assert.Equal(objectResult.StatusCode, 500);
             Assert.Equal(objectResult.Value, "Specialize MSI sidecar call failed. StatusCode=BadRequest");
         }
+
+        [Fact]
+        public async Task Disable_Restarts_ScriptHost()
+        {
+            var environment = new TestEnvironment();
+            environment.SetEnvironmentVariable(EnvironmentSettingNames.ContainerName, "container-name");
+            environment.SetEnvironmentVariable(EnvironmentSettingNames.AzureWebsiteInstanceId, string.Empty);
+
+            var loggerFactory = new LoggerFactory();
+            var loggerProvider = new TestLoggerProvider();
+            loggerFactory.AddProvider(loggerProvider);
+
+            var instanceController = new InstanceController(environment, null, loggerFactory);
+
+            var scriptHostManager = new Mock<IScriptHostManager>();
+
+            var fileSystem = new Mock<IFileSystem>();
+            var fileBase = new Mock<FileBase>();
+            fileBase.Setup(
+                    f => f.Exists(It.Is<string>(path => path.EndsWith(ScriptConstants.DisableContainerFileName))))
+                .Returns(true);
+            fileSystem.SetupGet(fs => fs.File).Returns(fileBase.Object);
+
+            FileUtility.Instance = fileSystem.Object;
+
+            scriptHostManager.Setup(s => s.RestartHostAsync(It.IsAny<CancellationToken>()));
+
+            var actionResult = await instanceController.Disable(scriptHostManager.Object);
+
+            FileUtility.Instance = null;
+
+            scriptHostManager.Verify(s => s.RestartHostAsync(It.IsAny<CancellationToken>()), Times.Once);
```

Comment:
Can you also verify the disable file mock to ensure the file was written?

Reply:
Added a assert on the contents of the stream.